glob/accessor_common.go

```diff
@@ -171,21 +172,32 @@ func (self OSFileSystemAccessor) New(scope vfilter.Scope) (FileSystemAccessor, e
 }
 
 func (self OSFileSystemAccessor) Lstat(filename string) (FileInfo, error) {
-	lstat, err := os.Lstat(self.GetPath(filename))
+	pathSpec, err := PathSpecFromString(filename)
+	if err != nil {
+		return nil, err
+	}
+
+	lstat, err := os.Lstat(path.Join(pathSpec.DelegatePath, pathSpec.Path))
 	if err != nil {
 		return nil, err
 	}
 
 	return &OSFileInfo{
 		_FileInfo:     lstat,
-		_full_path:    filename,
+		_full_path:    pathSpec.String(),
 		_accessor_ctx: self.context,
 	}, nil
 }
 
-func (self OSFileSystemAccessor) ReadDir(path string) ([]FileInfo, error) {
-	path = self.GetPath(path)
-	lstat, err := os.Lstat(path)
+func (self OSFileSystemAccessor) ReadDir(dir string) ([]FileInfo, error) {
+	pathSpec, err := PathSpecFromString(dir)
+	if err != nil {
+		return nil, err
+	}
+
+	fullpath := path.Join(pathSpec.DelegatePath, pathSpec.Path)
```

Comment:
> So you want to get the OSFileSystemAccessor to take a delegate pathspec so it can be rooted at any arbitrary accessor/path. This is a great idea but I think it is actually a completely different type of accessor - say we call it a MountedAccessor

I had not yet thought of a dedicated accessor for that but yes, this is essentially what I had in mind when I was playing around with different approaches for the implementation. Thinking about it, you may have just proposed a very good solution to the whole remapping thing because internally the `mount` accessor could just replace the `file` accessor if remappings are enabled and everything works out of the box.

Thinking two steps ahead, using pathspecs this way could perhaps also allow pathspec chaining (think of something like "read a file from a zip-compressed disk image"). I'm not sure if this would ever be needed, though, but even if, this would be something for the (far) future and imo there is other things which need to be set up first before things like chaining would make any sense.

Reply:
No this is exactly how it works :-). Pathspec chaining is already used to recursively search zip files - see eg
https://docs.velociraptor.app/exchange/artifacts/pages/log4jvulnhunter/

Reply:
Isn't that only implement for zips by now? I thought of chaining in a more general way which is afaict not yet possible because the other accessors cannot handle pathspecs for now.